server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionTaskManager.java

```diff
@@ -382,11 +382,20 @@ public void restart() throws InterruptedException {
         this.taskExecutionPool.shutdownNow();
         this.taskExecutionPool.awaitTermination(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
       }
+      if (subCompactionTaskExecutionPool != null) {
+        this.subCompactionTaskExecutionPool.shutdownNow();
+        this.subCompactionTaskExecutionPool.awaitTermination(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
```

Comment:
do not wait MAX_VALUE, wait some time then throw exception

Reply:
Resolved